### PR TITLE
#169817804:redux setup

### PR DESCRIPTION
**Pull request by mujohn26**
### What does this PR do?
- setting up redux
### Description of Task to be completed?
- installed redux dependencies
- added redux folder structure
### How should this be manually tested?
- git clone the repository by running `git clone  https://github.com/andela/blackninjas-frontend.git`
- run` npm i` to install dependencies
- run npm start to start the server

### What are the relevant pivotal tracker stories?
[#169817804](https://www.pivotaltracker.com/story/show/169817804)
### Screenshots (if appropriate)
Screen Shot 2020-01-29 at 12 47 29
Screen Shot 2020-01-29 at 12 47 49

### CODE REVIEW

File: src/redux/actionTypes/actionTypes.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const GREETING = 'GREETING';
```
Comment (Ruqoyah) [inline src/redux/actionTypes/actionTypes.js]: Change this file path to src/actions/actionTypes.js, there is no need for the redux folder

File: src/redux/actions/actions.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { GREETING,GOOD_BYE} from '../actionTypes/actionTypes';
```
Comment (Ruqoyah) [inline src/redux/actions/actions.js]: I'm not expecting us to have all action in one file. So you could change the name of this file to `/src/actions/homeAction.js` so everyone knows they need to follow the naming convention.

File: src/redux/reducers/reducer.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (Ruqoyah) [inline src/redux/reducers/reducer.js]: You have to take out the redux folder here too. I also don't expect all reducer to be on one file, so if the action is called homeAction.js, I expect you name this homeReducer.js, then create index.js where all the reducers created can be imported and combine using combineReducers from redux.

**Review by Ruqoyah — CHANGES_REQUESTED** [review]

File: src/redux/actionTypes/actionTypes.js (hunk 1 above)
Comment (mujohn26) [reply src/redux/actionTypes/actionTypes.js]: Done

File: src/redux/actions/actions.js (hunk 2 above)
Comment (mujohn26) [reply src/redux/actions/actions.js]: Done

File: src/redux/reducers/reducer.js (hunk 3 above)
Comment (mujohn26) [reply src/redux/reducers/reducer.js]: @Ruqoyah you mean to take the redux folder outside of the src folder? besides am done with creating and index.js with combineReducers.

File: src/components/index.jsx [hunk 4]
```diff
@@ -1,12 +1,15 @@
 import React, { Component } from 'react';
 import Welcome from './Welcome.jsx';
-
+import {Provider} from 'react-redux';
```
Comment (Ruqoyah) [inline src/components/index.jsx]: Please rename this file to App.jsx as said yesterday. Also, import the provider and store in src/index.js where the App component is called

**Review by Ruqoyah — CHANGES_REQUESTED** [review]

**Comment by Ruqoyah** [comment]
Also, please get rid of package-lock.json from your PR and ensure it's added appropriately to .gitignore

File: src/components/index.jsx (hunk 4 above)
Comment (mujohn26) [reply src/components/index.jsx]: I finished implementing this.

File: __tests__/welcome.test.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import App from '../src/components/App';
+import {Welcome} from '../src/components/Welcome';
+import {props} from '../__mockData__/redux-mock-data'
+describe('Render root app component', () => {
+    it('should render the root welcome component successfully', () => {
+        const wrapper = mount(<Welcome  {...props}/>);
+        expect(wrapper.text()).toEqual('Hello world To barefoot nomadGreetingGoodBye');
```
Comment (Ruqoyah) [inline __tests__/welcome.test.js]: This isn't a proper test, the page needs to be tested properly e.g `expect(wrapper.find('h1').text()).toEqual('welcome To barefoot nomad');` `expect(wrapper.find('h1').length).toBe(1);` and so on.

File: __tests__/App.test.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
-import App from '../src/components/index';
+import { mount, shallow } from 'enzyme';
+import App from '../src/components/App';
 
 describe('Render root app component', () => {
     it('should render the root app component successfully', () => {
-        const wrapper = mount(<App/>);
-        expect(wrapper.text()).toEqual('You are welcome at Barefoot nomad');   
+        const wrapper = shallow(<App/>);
+        expect(wrapper.text()).toEqual('');
```
Comment (Ruqoyah) [inline __tests__/App.test.js]: Just test the length of the wrapper instead of testing an empty string

**Review by Ruqoyah — CHANGES_REQUESTED** [review]

**Pull request merged**